### PR TITLE
Update for v12 compatibility

### PR DESCRIPTION
**Pull request by Aioros**
- on `canvasReady`, call `drawDrawingsMask` in an animation frame
- updated module manifest, removed older compatibility fields, removed manifestPlus field because I don't believe that's relevant anymore
- changed lingering references to `globalThis.randomID()` to `foundry.utils.randomID()`
- changed reference to `PIXI.filter.DisplacementFilter` to `PIXI.DisplacementFilter`

### CODE REVIEW

**Comment by Aioros** [comment]
I just noticed you use Conventional Commits syntax. Let me know if you want me to redo the PR after review.

**Comment by ghost91-** [comment]
Thanks! I’m currently on vacation and don’t have a laptop with me. I’ll take a look as soon as I’m back (11.08.2024).

**Comment by ThiefMaster** [comment]
Tip, you can simply `git commit --amend` to change the commit message and then force-push it. No need to create a new PR just to change the commit message :)

**Comment by ThiefMaster** [comment]
I think some more changes are needed: When dragging an effect from the "Special Effect Management" panel onto the scene, it creates a tile but it does not set the file path, so the tile remains empty (if I manually add the path to the tile afterwards, it's played correctly).

**Comment by ThiefMaster** [comment]
Looks like it's because of this API change:

> Tile Document
> Removed automatic migration of the v9 `TileDocument#img` field which was deprecated in v10 and renamed to `TileDocument#texture#src`.

Adding this change fixes it:


```patch
diff --git a/src/fxmaster.js b/src/fxmaster.js
index 7e32150..77db41a 100644
--- a/src/fxmaster.js
+++ b/src/fxmaster.js
@@ -151,7 +151,7 @@ Hooks.on("dropCanvasData", async (canvas, data) => {
     flags: {},
     height: data.height,
     hidden: false,
-    img: data.file,
+    texture: {src: data.file},
     locked: false,
     occlusion: { mode: 1, alpha: 0 },
     overHead: false,
```

Not sure if this one is relevant as well (width/height are set, but no idea if negative values are ever used):

> Removed automatic migration of assigning a negative TileDocument#width or TileDocument#height in favor of assigning a negative TileDocument#texture#scaleX and TileDocument#texture#scaleY respectively.

**Comment by Aioros** [comment]
I'll take a look at that too and see if I can add it, thanks. And thanks for the git tip too, God knows I still need them after all these years.

**Comment by Aioros** [comment]
Updated the PR with @ThiefMaster's contribution and a new commit message.

File: module.json [hunk 1]
```diff
@@ -29,10 +29,8 @@
   "version": "4.0.2",
   "compatibility": {
     "minimum": "11.299",
-    "verified": "11"
+    "verified": "12.329"
```
Comment (ThiefMaster) [inline module.json:32]:
```suggestion
    "verified": "12.330"
```

**Comment by Aioros** [comment]
Apologies for being _that guy_: just checking if you had any time to take a look at this PR.

**Comment by ghost91-** [comment]
No worries, I know the feeling ;)

I'm back from vacation now and I plan to do it this week.

File: src/fxmaster.js
Comment (dralois) [inline src/fxmaster.js:1]: The new line should be `texture: { src: data.file },`, otherwise lint will complain

**Review by dralois — CHANGES_REQUESTED** [review]
Overall it seems to work, one tiny change is required to make lint happy.

File: src/fxmaster.js [hunk 2]
```diff
@@ -151,7 +151,7 @@ Hooks.on("dropCanvasData", async (canvas, data) => {
     flags: {},
     height: data.height,
     hidden: false,
-    img: data.file,
+    texture: {src: data.file},
```
Comment (dralois) [inline src/fxmaster.js]: This change is required otherwise lint will complain
```suggestion
    texture: { src: data.file },
```

**Review by dralois — APPROVED** [review]

**Review by dralois — APPROVED** [review]

**Review by dralois — APPROVED** [review]

**Comment by ghost91-** [comment]
Merged via #765. Thanks again for your work!

**Pull request closed without merging**